### PR TITLE
Adding 3 to python shebangs

### PR DESCRIPTION
**Pull request by bsb808**
Be explicit with which version of python scripts should use.  Discovered as part of PR review: https://github.com/Field-Robotics-Lab/dave/pull/128

### CODE REVIEW

**Review by M1chaelM — CHANGES_REQUESTED** [review]
While we are changing the shebangs it would also be good to check that the code is Python3 compatible. A quick way to do this is to run
```
pylint --py3k <script_to_check.py>
```
on each of the Python files in the repo and address the errors. The total list of Python files in this repo is:
```
./uuv_manipulators_control/src/uuv_manipulators_control/__init__.py
./uuv_manipulators_control/src/uuv_manipulators_control/cartesian_controller.py
./uuv_manipulators_control/setup.py
./uuv_manipulators_control/scripts/jt_cartesian_controller.py
./uuv_manipulators_control/scripts/joint_position_controller.py
./uuv_manipulators_control/scripts/gripper_controller.py
./uuv_manipulators_control/scripts/set_joint_config.py
./uuv_manipulators_kinematics/test/test_arm_interface.py
./uuv_manipulators_kinematics/src/uuv_kinematics_utils/__init__.py
./uuv_manipulators_kinematics/src/uuv_kinematics_utils/endeffector_state.py
./uuv_manipulators_kinematics/src/uuv_manipulator_nodes/__init__.py
./uuv_manipulators_kinematics/src/uuv_manipulator_nodes/endpoint_state.py
./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/__init__.py
./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/arm.py
./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/kin_chain.py
./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/gripper.py
./uuv_manipulators_kinematics/setup.py
./uuv_manipulators_kinematics/scripts/kinematics_service.py
```
I've gotten through everything under `manipulators_control` but I'm out of time for now. It would be great if someone could do `manipulators_kinematics`.

**Comment by crvogt** [comment]
> While we are changing the shebangs it would also be good to check that the code is Python3 compatible. A quick way to do this is to run
> 
> ```
> pylint --py3k <script_to_check.py>
> ```
> 
> on each of the Python files in the repo and address the errors. The total list of Python files in this repo is:
> 
> ```
> ./uuv_manipulators_control/src/uuv_manipulators_control/__init__.py
> ./uuv_manipulators_control/src/uuv_manipulators_control/cartesian_controller.py
> ./uuv_manipulators_control/setup.py
> ./uuv_manipulators_control/scripts/jt_cartesian_controller.py
> ./uuv_manipulators_control/scripts/joint_position_controller.py
> ./uuv_manipulators_control/scripts/gripper_controller.py
> ./uuv_manipulators_control/scripts/set_joint_config.py
> ./uuv_manipulators_kinematics/test/test_arm_interface.py
> ./uuv_manipulators_kinematics/src/uuv_kinematics_utils/__init__.py
> ./uuv_manipulators_kinematics/src/uuv_kinematics_utils/endeffector_state.py
> ./uuv_manipulators_kinematics/src/uuv_manipulator_nodes/__init__.py
> ./uuv_manipulators_kinematics/src/uuv_manipulator_nodes/endpoint_state.py
> ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/__init__.py
> ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/arm.py
> ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/kin_chain.py
> ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/gripper.py
> ./uuv_manipulators_kinematics/setup.py
> ./uuv_manipulators_kinematics/scripts/kinematics_service.py
> ```
> 
> I've gotten through everything under `manipulators_control` but I'm out of time for now. It would be great if someone could do `manipulators_kinematics`.

@M1chaelM  Checking it

**Comment by bsb808** [comment]
> > While we are changing the shebangs it would also be good to check that the code is Python3 compatible. A quick way to do this is to run
> > ```
> > pylint --py3k <script_to_check.py>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > on each of the Python files in the repo and address the errors. The total list of Python files in this repo is:
> > ```
> > ./uuv_manipulators_control/src/uuv_manipulators_control/__init__.py
> > ./uuv_manipulators_control/src/uuv_manipulators_control/cartesian_controller.py
> > ./uuv_manipulators_control/setup.py
> > ./uuv_manipulators_control/scripts/jt_cartesian_controller.py
> > ./uuv_manipulators_control/scripts/joint_position_controller.py
> > ./uuv_manipulators_control/scripts/gripper_controller.py
> > ./uuv_manipulators_control/scripts/set_joint_config.py
> > ./uuv_manipulators_kinematics/test/test_arm_interface.py
> > ./uuv_manipulators_kinematics/src/uuv_kinematics_utils/__init__.py
> > ./uuv_manipulators_kinematics/src/uuv_kinematics_utils/endeffector_state.py
> > ./uuv_manipulators_kinematics/src/uuv_manipulator_nodes/__init__.py
> > ./uuv_manipulators_kinematics/src/uuv_manipulator_nodes/endpoint_state.py
> > ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/__init__.py
> > ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/arm.py
> > ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/kin_chain.py
> > ./uuv_manipulators_kinematics/src/uuv_manipulator_interfaces/gripper.py
> > ./uuv_manipulators_kinematics/setup.py
> > ./uuv_manipulators_kinematics/scripts/kinematics_service.py
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I've gotten through everything under `manipulators_control` but I'm out of time for now. It would be great if someone could do `manipulators_kinematics`.
> 
> @M1chaelM Checking it

Thank you guys!

**Comment by crvogt** [comment]
Finished the check on the remaning files. In the case of the `range()` calls, I cast the output to a list with `list(range())`, as per the Python2 functionality, but not sure if this is best practice.

**Comment by bsb808** [comment]
> Finished the check on the remaning files. In the case of the `range()` calls, I cast the output to a list with `list(range())`, as per the Python2 functionality, but not sure if this is best practice.

Awesome - thank you @crvogt

**Comment by bsb808** [comment]
I'll merge when we get an approval.

**Comment by crvogt** [comment]
> I'll merge when we get an approval.

@bsb808 Do we want to assign someone who hasn't altered any code, or do you want me to go ahead?

**Comment by M1chaelM** [comment]
> Finished the check on the remaning files. In the case of the `range()` calls, I cast the output to a list with `list(range())`, as per the Python2 functionality, but not sure if this is best practice.

It's not, but it won't make it worse and if the range wasn't very big it shouldn't matter much. If we can get away with just the range that would be a little better.

**Comment by crvogt** [comment]
@M1chaelM I wasn't sure how to adjust ` range()` based on the `pylint` output. I think it's a similar situation to calling dictionary keys. If we're not specifically trying to get rid of `pylint` errors, I can go back and remove the `list` calls.

File: uuv_manipulators_kinematics/scripts/kinematics_service.py
Comment (M1chaelM) [inline uuv_manipulators_kinematics/scripts/kinematics_service.py]: Let's leave this for now. It has to be a list to work with zip and I doubt there are millions of joint_names...

File: uuv_manipulators_kinematics/test/test_arm_interface.py
Comment (M1chaelM) [inline uuv_manipulators_kinematics/test/test_arm_interface.py]: Same for this one and below.

**Review by M1chaelM — APPROVED** [review]
All looks good to me! I hadn't looked closely at the range-related changes but I think they are fine.

File: uuv_manipulators_kinematics/scripts/kinematics_service.py
Comment (M1chaelM) [reply uuv_manipulators_kinematics/scripts/kinematics_service.py]: oops, my mistake. zip will work with a range object. Not a big deal but I think slightly better/more readable to use enumerate. I'll push this now.

**Comment by crvogt** [comment]
@M1chaelM Ah thanks, completely blanked on enumerate!

**Pull request merged**
